scrapper.py

```diff
@@ -25,39 +36,121 @@ class Crawler:
     """
     Crawler implementation
     """
+
     def __init__(self, seed_urls, max_articles: int):
-        pass
+        self.seed_urls = seed_urls
+        self.max_articles = max_articles
+        self.urls = []
 
     def _extract_url(self, article_bs):
-        pass
+        content_bs = article_bs.find_all('div', class_="issueArticle flex")
+        for tag in content_bs:
+            link = tag.find('a')
+            main_link = link['href']
+            self.urls.append(main_link)
 
     def find_articles(self):
         """
         Finds articles
         """
-        pass
+        for seed_url in self.seed_urls:
+            response = requests.get(seed_url, HEADERS)
+            article_bs = BeautifulSoup(response.text, 'html.parser')
+            self._extract_url(article_bs)
 
     def get_search_urls(self):
         """
         Returns seed_urls param
         """
-        pass
+        return self.seed_urls
 
 
 def prepare_environment(base_path):
     """
     Creates ASSETS_PATH folder if not created and removes existing folder
     """
-    pass
+    if not os.path.exists(base_path):
+        os.makedirs(base_path)
 
 
 def validate_config(crawler_path):
     """
     Validates given config
     """
-    pass
+    with open(crawler_path, 'r', encoding='utf-8') as config:
+        scrapper_config = json.load(config)
+
+    seed_urls = scrapper_config["seed_urls"]
+    total_articles_to_find_and_parse = scrapper_config["total_articles_to_find_and_parse"]
+
+    if not isinstance(seed_urls, list):
+        raise IncorrectURLError
+    if not seed_urls:
+        raise IncorrectURLError
+    for seed_url in seed_urls:
+        if not re.match('https://', seed_url):
+            raise IncorrectURLError
+
+    if not isinstance(total_articles_to_find_and_parse, int) or total_articles_to_find_and_parse <= 0:
+        raise IncorrectNumberOfArticlesError
+
+    if total_articles_to_find_and_parse > 300:
+        raise NumberOfArticlesOutOfRangeError
+
+    return seed_urls, total_articles_to_find_and_parse
+
+
+class HTMLParser:
+    """
+    Parser implementation
+    """
+
+    def __init__(self, article_url, article_id):
+        """
+        Init
+        """
+        self.article_url = article_url
+        self.article_id = article_id
+        self.article = Article(url=article_url, article_id=article_id)
+
+    def _fill_article_with_text(self, article_bs):
+
+        table_rows = article_bs.find('div', class_="fulltext")
+        link = table_rows.find('a')['href']
+        new_link = re.sub(r'(?i)view(?=\W)', 'download', link)
+        pdf_file = PDFRawFile(new_link, self.article_id)
+        pdf_file.download()
+        self.article.text = pdf_file.get_text()
+        self.article.save_raw()
```

Comment:
do not break the SRA principle: https://en.wikipedia.org/wiki/Single-responsibility_principle

Reply:
it should not be saved in the place where we get the text.